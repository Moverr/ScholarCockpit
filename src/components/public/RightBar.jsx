import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { logTable } from '../../utils/Logger';


import '../../css/nav.css';
import Banner from './Banner';
import VerifySchool from './VerifySchool';
import Login from './Login';




const logdata = (
  {
    "section": "HeaderBar",
    "message": "Layout to the header section ",
    "props": "N/A",

  }
);

const renderSwitch = (level, props) => {

  switch (level) {
    case 'LOGIN':
      return <Login props={props} />;
      

    case 'VERIFYSCHOOL':
      return <VerifySchool props={props} />;
     

    default:
      return <VerifySchool props={props} />;
     

  }
}

function RightBar(props) {

  const screennavigation = useSelector(state => state.navigation);

  logTable(logdata);
  const urlsegment = props.urlSegment;
  console.log("SOEOEOEOEOEOEO");
  console.log(screennavigation.currentScreen);


  return (
    <div className={"row-fluid"}>
      <div className="row headerbar">

        <div className="col-md-9">
          <Banner props={props} />
        </div>
        <div className="col-md-3">
          {renderSwitch(screennavigation.currentScreen, props)}
        </div>



      </div>

    </div>

  )

}
export default RightBar;
