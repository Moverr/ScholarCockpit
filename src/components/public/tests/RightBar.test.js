import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import RightBar from '../RightBar';
import { Provider } from 'react-redux';






let wrapper = shallow( <RightBar />);



describe('Testing Rightbar ', () => {


  it('should  render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have a class banner', () => {
    expect(wrapper).toMatchSnapshot();
  });
})



