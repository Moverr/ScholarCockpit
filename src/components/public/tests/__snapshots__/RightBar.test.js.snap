// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rightbar  Should have a class banner 1`] = `undefined`;

exports[`Testing Rightbar  should  render correctly 1`] = `undefined`;
